roles/glance/tasks/main.yml

```diff
@@ -71,14 +71,23 @@
     - glance-registry
   when: os == 'ubuntu'
 
-- name: install glance services
+- name: install glance services (rhel source)
   systemd_service: name={{ item }}
                    user=glance
                    cmd=/usr/local/bin/{{ item }}
   with_items:
     - glance-api
     - glance-registry
-  when: os == 'rhel'
+  when: os == 'rhel' and openstack_install_method == 'source'
+  
+- name: install glance services (rhel source)
+  systemd_service: name={{ item }}
```

Comment:
Oh, the Ursula I used also include this task, so I think that unit file was also created by this task, may not auto created by glance.